include/cosim/system_structure.hpp

```diff
@@ -213,8 +213,47 @@ class system_structure
         full_variable_name target;
     };
 
+    /// Information about a powerbond connection. For use with the ecco algorithm only.
+    struct power_bond
+    {
+        /// The u_a variable in the bond, according to the description of the ecco algorithm.
+        full_variable_name u_a{"", ""};
+
+        /// The u_b variable in the bond, according to the description of the ecco algorithm.
+        full_variable_name u_b{"", ""};
+
+        /// The y_a variable in the bond, according to the description of the ecco algorithm.
+        full_variable_name y_a{"", ""};
+
+        /// The y_b variable in the bond, according to the description of the ecco algorithm.
+        full_variable_name y_b{"", ""};
+
+        void set_ua(full_variable_name& ua)
+        {
+            u_a = ua;
+        }
+
+        void set_ub(full_variable_name& ub)
+        {
+            u_b = ub;
+        }
+
+        void set_ya(full_variable_name& ya)
+        {
+            y_a = ya;
+        }
+
+        void set_yb(full_variable_name& yb)
+        {
+            y_b = yb;
+        }
+    };
+
+    std::string algorithm;
```

Comment:
`system_structure` is only supposed to describe the modelled *system*; I don't think it should contain knowledge/requirements concerning the algorithm. (It is fine to have the info about the power bonds here, though, just document that not all algorithms will use it.)

Reply:
Added it reflexively, but it wasn't actually used anywhere in the code yet. So removed.

Reply:
I was more thinking of `system_structure `as a representation of the `OspSystemStructure`. 

If we don't add it here, we'll need to implement a different way of exposing which algorithm is defined in the XML.